specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2023-06-01-preview/Migrations.json

```diff
@@ -0,0 +1,1254 @@
+        "Canceled",
+        "Failed",
+        "Succeeded",
+        "ValidationFailed",
+        "CleaningUp"
+      ],
+      "type": "string",
+      "description": "Migration state.",
+      "x-ms-enum": {
+        "name": "MigrationState",
+        "modelAsString": true
+      }
+    },
+    "MigrationStatus": {
+      "type": "object",
+      "properties": {
+        "state": {
+          "$ref": "#/definitions/MigrationState",
+          "readOnly": true,
+          "description": "State of migration"
+        },
+        "error": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Error message, if any, for the migration state"
+        },
+        "currentSubStateDetails": {
+          "$ref": "#/definitions/MigrationSubStateDetails",
+          "readOnly": true,
+          "description": "Current Migration sub state details."
+        }
+      },
+      "description": "Migration status."
+    },
+    "MigrationSubState": {
+      "enum": [
+        "PerformingPreRequisiteSteps",
+        "WaitingForLogicalReplicationSetupRequestOnSourceDB",
+        "WaitingForDBsToMigrateSpecification",
+        "WaitingForTargetDBOverwriteConfirmation",
+        "WaitingForDataMigrationScheduling",
+        "WaitingForDataMigrationWindow",
+        "MigratingData",
+        "WaitingForCutoverTrigger",
+        "CompletingMigration",
+        "Completed",
+        "CancelingRequestedDBMigrations",
+        "ValidationInProgress"
+      ],
+      "type": "string",
+      "description": "Migration sub state.",
+      "x-ms-enum": {
+        "name": "MigrationSubState",
+        "modelAsString": true
+      }
+    },
+    "MigrationSubStateDetails": {
+      "type": "object",
+      "properties": {
+        "currentSubState": {
+          "$ref": "#/definitions/MigrationSubState",
+          "readOnly": true
+        },
+        "dbDetails": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/DbMigrationStatus"
+          }
+        },
+        "validationDetails": {
+          "$ref": "#/definitions/ValidationDetails"
+        }
+      },
+      "description": "Migration sub state details."
+    },
+    "DbMigrationStatus": {
+      "type": "object",
+      "description": "Migration status of an individual database",
+      "properties": {
+        "databaseName": {
+          "type": "string",
+          "description": "Name of the database"
+        },
+        "migrationState": {
+          "type": "string",
+          "description": "Migration db state of an individual database",
+          "enum": [
+            "InProgress",
+            "WaitingForCutoverTrigger",
+            "Failed",
+            "Canceled",
+            "Succeeded",
+            "Canceling"
+          ],
+          "x-ms-enum": {
+            "name": "MigrationDbState",
+            "modelAsString": true
+          }
+        },
+        "migrationOperation": {
+          "type": "string",
```

Comment:
enum?

Reply:
migrationOperation purpose is to explain to the customer what is currently going on in the migration. 
for eg:
Creating Roles
Cleaning Target tb
Dropping database
Migrating data
etc
There can be multiple steps in the migration like the above hence we have kept it as string.
 We have another property that you see in this file called "MigrationState". This we have kept as enum and the values in this property will be (Inprogress, Canceling, Failed, Succeeded , WaitingForCutover)

Below is how it will be show to the customer:
MigrationState:Inprogress
MigrationOperation:Creating Roles

MigrationState:Inprogress
MigrationOperation:Dropping database

etc

Reply:
Since there can be multiple operations involved during migration, creating an enum will not work in this case. Also this being read-only user will not gain any value from this being an enum. Hence will mark as By Design.

Reply:
treating this as a description field and approving this